### ISSUE TITLE
Compilation fails on tsc versions >= 4.2.0

### ISSUE THREAD

**Issue by rrichardson** [grissius/protocat#37]
When running on 4.2.0 or greater: 

```
src/lib/misc/type-helpers.ts:20:39 - error TS2344: Type 'KnownKeys<T>' does not satisfy the constraint 'keyof T'.
  Type 'unknown' is not assignable to type 'keyof T'.

20 export type RemoveIdxSgn<T> = Pick<T, KnownKeys<T>>
                                         ~~~~~~~~~~~~

src/lib/misc/type-helpers.ts:22:40 - error TS2344: Type 'NeverKeys<T>' does not satisfy the constraint 'keyof T'.
  Type 'unknown' is not assignable to type 'keyof T'.

22 export type OmitNeverKeys<T> = Pick<T, NeverKeys<T>>
                                          ~~~~~~~~~~~~

src/lib/server/call.ts:129:9 - error TS2322: Type 'KnownKeys<{ [M in keyof T]: MethodDef2ServiceHandler<T[M], Extension>; }>' is not assignable to type 'string | number | symbol'.
  Type 'unknown' is not assignable to type 'string | number | symbol'.
    Type 'unknown' is not assignable to type 'symbol'.
      Type 'KnownKeys<{ [M in keyof T]: MethodDef2ServiceHandler<T[M], Extension>; }>' is not assignable to type 'symbol'.
        Type 'unknown' is not assignable to type 'symbol'.

129   [M in keyof ServiceImplementation<T, Extension>]:
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by grissius** [comment]
Thanks for reporting, will try to fix within the following days.

**Comment by grissius** [comment]
Thanks for reporting, fix is on the way. I could not find the breaking change in TS release notes, so I reported an issue for that: https://github.com/microsoft/TypeScript/issues/43357.

**Comment by grissius** [comment]
Released with `0.6.2`
